content/articles/fcm-with-spring-and-angular/index.md

```diff
@@ -0,0 +1,500 @@
+Title: Sending Notifications with Spring Boot, Angular, and Firebase Cloud Messaging
+
+Description: A starting guide on sending notifications to an Angular application. To do this, we will be using a Spring Boot backend and Firebase Cloud Messaging.
+
+### Introduction
+
+Notifications are a great way to increase user engagement. By keeping them notified of events in your app that interest them, you can keep them coming back to you. Many apps even rely on notifications as a core feature. For example, there are some reminder apps that help you remember important events. What would they be without being able to notify you while inactive? Either way, sending notifications is an important skill to learn as a developer.
+
+In this guide, we will be learning how to send notifications to an Angular application. To make this possible, we will be using a Spring Boot backend with the help of **Firebase Cloud Messaging**. By the end of this guide, you should have a good understanding of how to send notifications in your next full-stack application.
+
+### Table of Contents
+
+- [Understanding the high-level architecture of our project](#understanding-the-high-level-architecture-of-our-project)
+- [Setting up our backend](#setting-up-our-backend)
+- [Creating a Firebase Cloud Messaging service](#creating-a-firebase-cloud-messaging-service)
+- [Exposing our service through a REST controller](#exposing-our-service-through-a-rest-controller)
+- [Setting up Firebase on our front-end](#setting-up-firebase-on-our-front-end)
+- [Requesting permission to send notifications](#requesting-permission-to-send-notifications)
+- [Subscribing to Receive notifications](#subscribing-to-receive-notifications)
+- [Conclusion](#conclusion)
+
+### Prerequisites
+
+For this guide, you will need basic knowledge of Angular and Spring Boot since that will be our tech stack. For our backend, we will be using Kotlin as our JVM language. Yet, this guide should still be easy to follow as a Java developer. Additionally, you won’t need to know any complex Spring Boot concepts since they are not needed here. You will only need to know Spring MVC and the basic design patterns (i.e. [beans](https://www.baeldung.com/spring-bean), and [stereotype annotations](https://medium.com/javarevisited/spring-stereotype-annotations-1469ca0c3ad2)). Although there may be some small Spring Boot concepts here a beginner may not know. In which case, I’ll give an explanation along the way.
+
+On the front-end, you will more or less just need some basic angular knowledge to follow along. This includes using the CLI, the HTTP client, and basic templating. When it comes to Firebase, I won’t be expecting you to have any prior knowledge of how to use it. We will be going through all you need to know to use it for this situation. 
```

Comment:
Break down these paragraphs into bulleted-points.

The readers always find it easy to read points than to read bigger paragraphs.